src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -147,20 +349,82 @@ def forward(
         cache_position: Optional[torch.LongTensor] = None,
         position_embeddings: Optional[Tuple[torch.Tensor, torch.Tensor]] = None,  # will become mandatory in v4.46
     ):
+        bsz, seq_len, _ = hidden_states.shape
+
         # Apply attention-specific projections and rope
         query, key, value, present = self._attn_projections_and_rope(
             hidden_states=hidden_states,
             position_ids=position_ids,
             layer_past=layer_past,
             use_cache=use_cache,
+            cache_position=cache_position,
             position_embeddings=position_embeddings,
         )
 
+        # Flash Attention 2 specific handling for PEFT integration
+        target_dtype = None
+        if self.config._attn_implementation == "flash_attention_2":
+            input_dtype = query.dtype
+            if input_dtype == torch.float32:
+                if torch.is_autocast_enabled():
+                    target_dtype = torch.get_autocast_gpu_dtype()
+                # Handle the case where the model is quantized
+                elif hasattr(self.config, "_pre_quantization_dtype"):
+                    target_dtype = self.config._pre_quantization_dtype
+                else:
+                    target_dtype = self.query_key_value.weight.dtype
+
+        # Checking for fallbacks in case an unsupported feature is requested
+        attention_type = self.config._attn_implementation
+        if (output_attentions or head_mask is not None) and self.config._attn_implementation in [
+            "sdpa",
+            "flash_attention_2",
+        ]:
+            warning_msg = "Setting `attention_type` to `eager` because"
+
+            if output_attentions:
+                warning_msg += " `output_attentions=True`"
+            if output_attentions and head_mask is not None:
+                warning_msg += " and `head_mask` not None"
+            elif head_mask is not None:
+                warning_msg += " `head_mask` not None"
+            warning_msg += f" not supported in {attention_type}"
+
+            logger.warning_once(warning_msg)
+            attention_type = "eager"
```

Comment:
That's supper verbose 😢 we could just say that we are falling back to eager due to config missmatch or something!

Reply:
Yea uh idk what I was thinking; it's ugly af xD changed it up!